route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <ros/ros.h>
+#include <string>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include "route_following_plugin.h"
+
+
+namespace route_following_plugin
+{
+    RouteFollowingPlugin::RouteFollowingPlugin() : mvr_duration_(16.0), current_speed_(0.0) { }
+
+    void RouteFollowingPlugin::initialize()
+    {
+        
+        nh_.reset(new ros::CARMANodeHandle());
+        pnh_.reset(new ros::CARMANodeHandle("~"));
+        
+        plan_maneuver_srv_ = nh_->advertiseService("plugins/RouteFollowingPlugin/plan_maneuvers", &RouteFollowingPlugin::plan_maneuver_cb, this);
```

Comment:
Should this just advertise `strategic_plan/plan_maneuvers` or `~/strategic_plan/plan_maneuvers` (I can never recall which one is which) so that it can be remapped in the launch file to the appropriate place?

Reply:
Sure.